src/server/controllers/devices.ts

```diff
@@ -42,6 +42,13 @@ const DeviceControler = (socket: Socket, io: Server) => ({
     const devices = devicesDb.getAll();
     socket.emit(EventTypes.DEVICES.SET, devices);
   },
+  deleteDevice: (deviceId: string) => {
+    devicesDb.delete(deviceId);
+
+    const devices = devicesDb.getAll();
+
+    socket.emit(EventTypes.DEVICES.SET, devices);
+  },
```

Comment:
To not leave things behind while deleting the device we want to delete all keys and all pages related to the device, starting from the smallest:

1. delete keys
2. delete pages
3. delete device